src/sentry_core.c

```diff
@@ -506,15 +506,6 @@ sentry__prepare_transaction(const sentry_options_t *options,
 {
     sentry_envelope_t *envelope = NULL;
 
-    bool should_skip = !sentry_value_is_true(
-        sentry_value_get_by_key(transaction, "sampled"));
-    if (should_skip) {
-        SENTRY_DEBUG("throwing away transaction due to sample rate");
-        goto fail;
-    }
-    // Field is superfluous, strip so it doesn't leak into the payload
-    sentry_value_remove_by_key(transaction, "sampled");
-
```

Comment:
this is taken care of earlier in the pipeline: see `sentry_transaction_finish`'s diff below, which performs the same check.